src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -19,13 +27,26 @@ DlgPrefWaveform::DlgPrefWaveform(
     setupUi(this);
 
     // Waveform overview init
-    waveformOverviewComboBox->addItem(tr("Filtered")); // "0"
-    waveformOverviewComboBox->addItem(tr("HSV")); // "1"
-    waveformOverviewComboBox->addItem(tr("RGB")); // "2"
-    m_pTypeControl = std::make_unique<ControlObject>(
-            ConfigKey(QStringLiteral("[Waveform]"),
-                    QStringLiteral("WaveformOverviewType")));
+    waveformOverviewComboBox->addItem(
+            tr("Filtered"), QVariant::fromValue(WOverview::Type::Filtered));
+    waveformOverviewComboBox->addItem(tr("HSV"), QVariant::fromValue(WOverview::Type::HSV));
+    waveformOverviewComboBox->addItem(tr("RGB"), QVariant::fromValue(WOverview::Type::RGB));
+    m_pTypeControl = std::make_unique<ControlPushButton>(kOverviewTypeCfgKey);
+    m_pTypeControl->setStates(QMetaEnum::fromType<WOverview::Type>().keyCount());
     m_pTypeControl->setReadOnly();
+    // Update the control with the config value
+    WOverview::Type overviewType =
+            m_pConfig->getValue<WOverview::Type>(kOverviewTypeCfgKey, WOverview::Type::RGB);
+    int cfgTypeIndex = waveformOverviewComboBox->findData(QVariant::fromValue(overviewType));
+    if (cfgTypeIndex == -1) {
+        // Invalid config value, set default type RGB and write it to config
+        waveformOverviewComboBox->setCurrentIndex(2);
```

Comment:
Could be worth using `WOverview::Type::RGB` instead of `2` not to have it hardcoded here

Reply:
Oh, indeed, I overlooked that

Reply:
I actually don't agree with this change. This conflates the value of the enum members with the order in which they are inserted. Since the hardcoded index in here is very close to the order in which they are inserted, its fine like this IMO.

Reply:
I already switched to the enum, but apparently that got lost somehow. Will fix again

Reply:
Oh, I confused it with resetToDefaults
fixed.